### ISSUE TITLE
OCR: PARSE FAILURE for images

### ISSUE THREAD

**Issue by mikkelnl** [the-paperless-project/paperless#586]
I've read several similar issues here, but can't find a solution, maybe someone can help?

I am running Paperless on a Synology in a docker container (so the containers for the CONSUMER and the SERVER are made via the Synology GUI).

When I try to let Paperless index a photo of a receipt with some Dutch language/text, it will eventually show this in the LOG:

`PARSE FAILURE for /consume/IMG_0855.jpg: The guessed language (so) is not available in this instance of Tesseract.`

or:

`PARSE FAILURE for /consume/IMG_0855.jpg: The guessed language (pl) is not available in this instance of Tesseract.`

or:

`PARSE FAILURE for /consume/IMG_0855.jpg: The guessed language (tl) is not available in this instance of Tesseract.`

I have set the following in the container environement:

PAPERLESS_OCR_LANGUAGES = nld eng

When I enter the command `tesseract --list-langs` I get:

List of available languages (6):                                                                                                                                              
dan                                                                                                                                                                           
deu                                                                                                                                                                           
eng                                                                                                                                                                           
equ                                                                                                                                                                           
nld                                                                                                                                                                           
osd 

Any ideas how to get Paperless to OCR an image with Dutch text?

**Comment by sceiler** [comment]
There are 2 environment variables.

- PAPERLESS_OCR_LANGUAGES
- PAPERLESS_OCR_LANGUAGE

I set both to my language and it works fine. Try this.

**Comment by mikkelnl** [comment]
> There are 2 environment variables.
> 
> * PAPERLESS_OCR_LANGUAGES
> * PAPERLESS_OCR_LANGUAGE
> 
> I set both to my language and it works fine. Try this.

Thanks, now it does 'consume' the image, but the content is not OCR'ed correctly; a lot of jibberish... :S

**Comment by sceiler** [comment]
Did it recognize the correct language? Otherwise you need to make sure that your scans are straight and the quality is good enough. I scan my documents with around 300dpi and usually in black/white.
But don't expect OCR to be perfect etc.
I personally would use a good title and tags to find documents instead of relying completely on OCR.

**Comment by mikkelnl** [comment]
Yeah, most my docs are good scans etc, but sometimes I have a receipt or small ticket of which I make a photo...

By adding both env variables you mentioned, and setting them to 'nld', I now have this in the logs:

OCRing the document                                                                                                                                                           
Parsing for nld                                                                                                                                                               
Language detection error: No features in text.                                                                                                                                
Language detection failed!                                                                                                                                                    
As FORGIVING_OCR is enabled, we're going to make the best with what we have.  

And when I look at the document, it did not OCR anything, contents is empty.... not sure what is wrong here....

**Comment by mikkelnl** [comment]
Never mind, it really was a bad scan. I noticed that especially the contrast is important. Good to know ;-)
